### PR TITLE
Add recall metrics to videoCLIP lightning module

### PR DESCRIPTION
**Pull request by sophiazhi**
Summary:
Add top-k recall and reverse recall metrics to VideoCLIP lightning module, with logging.

Test plan:
Compared my recall function to MUGEN's recall function to check for equal outputs ([notebook](https://colab.research.google.com/drive/190ZD0NadfXTI0OUWO-KVthP8zN1tniyq))
Calculated MUGEN's recall outputs on my mini test set of 100 samples ([notebook](https://colab.research.google.com/drive/10pXZZ9Nz5vy5SC237Brhy1pd5XZS4XNc))
Confirmed that my recall outputs matches MUGEN's after adding recall to the lightning module ([notebook](https://colab.research.google.com/drive/1uS8w0TIsc4hsGLBDdzxif97pAr9uPi6y))

### CODE REVIEW

File: examples/mugen/retrieval/model.py [hunk 1]
```diff
@@ -4,30 +4,37 @@
...
+from typing import Any, Tuple
+
+import torch
 
 from examples.mugen.retrieval.video_clip import videoclip
 from pytorch_lightning import LightningModule
+from torchmetrics import Recall
+
 from torchmultimodal.modules.losses.contrastive_loss_with_temperature import (
     ContrastiveLossWithTemperature,
```
Comment (langong347) [inline examples/mugen/retrieval/model.py:17]: it's not used anymore?

File: examples/mugen/retrieval/model.py [hunk 2]
```diff
@@ -4,30 +4,37 @@
...
 )
 
 
 class VideoCLIPLightningModule(LightningModule):
     """PyTorch Lightning module for evaluating VideoCLIP model.
-
     Args:
         logit_scale (float): Initial log-temperature value for contrastive loss funtion.
-            Defaults to ``3.8282``, MUGEN's log-temperature value after training.
+            Defaults to ``0.07``, MUGEN's log-temperature value at initialization.
```
Comment (langong347) [inline examples/mugen/retrieval/model.py:25]: Where did the ``3.8282`` come from previously? Why "after training" --> "at initialization"?

**Comment by langong347** [comment]
Did you update the last [notebook](https://colab.research.google.com/drive/1uS8w0TIsc4hsGLBDdzxif97pAr9uPi6y)? For recall I only saw a type error thrown in the end:

```
from pytorch_lightning import Trainer

trainer = Trainer()
trainer.test(pl_videoclip, dataloaders=datamodule.test_dataloader())
```

File: examples/mugen/retrieval/model.py (hunk 2 above)
Comment (sophiazhi) [reply examples/mugen/retrieval/model.py:25]: 3.8282 came from the checkpoint as the trained temperature parameter, but mugen initialized temperature to 0.07 so i think it makes more sense for the lightning module to initialize to 0.07 and only use 3.8282 when using the checkpoint

File: examples/mugen/retrieval/model.py (hunk 1 above)
Comment (sophiazhi) [reply examples/mugen/retrieval/model.py:17]: it's not, since i replaced measuring loss during evaluation to measuring recall (i kept the import and args because it will be used when i add training code). i think loss isn't usually reported during evaluation

**Review by RdoubleA — APPROVED** [review]
